common/src/main/java/io/pravega/common/io/filesystem/FileModificationMonitor.java

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.common.io.filesystem;
+
+import lombok.NonNull;
+
+import java.io.FileNotFoundException;
+import java.nio.file.Path;
+import java.util.function.Consumer;
+
+/**
+ * Represents an object that monitors modifications to a file.
+ *
+ * By modification, we mean:
+ * - Edits to an existing file
+ * - Replacement of the whole file
+ */
+public interface FileModificationMonitor {
+
+    /**
+     * Start monitoring.
+     */
+    void startMonitoring();
+
+    /**
+     * Stop monitoring.
+     */
+    void stopMonitoring();
+
+    /**
+     * A convenience method for implementations.
+     *
+     * @param filePath the path of the file to be monitored
+     * @param callback the callback to be invoked when a file modification is detected
```

Comment:
1. Why is this argument here? it's not used. If all you need to do is check for nulls, then annotate with `@NonNull` wherever you are accepting it.

2. Move this method in the implementation. There is no need for it to be exposed to the outside as it doesn't provide any meaningful use.

Reply:
Yes, that makes sense. It was looking awkward to me too, but I somehow missed modifying it.